definitions/sources/declares.js

```diff
@@ -0,0 +1,26 @@
+const staging_tables = ["pages", "requests", "parsed_css"]
```

Comment:
Is this `declare.js` file used anywhere?

Reply:
Yes, it generates the original data sources.
See FROM statements in all/pages.js or core_web_vital/technologies.js

This is where we would also put asserts before running after_crawl pipeline.

Reply:
Ah got you. Missed that.

> This is where we would also put asserts before running after_crawl pipeline.

Was curious about when that assert runs! How is it defined that's where they run?

Reply:
We define it in the graph code: in `core_web_vitals/technologies.js` file as `dependOnDependencyAssertions: true`.
![Screenshot 2024-08-29 at 20 30 27](https://github.com/user-attachments/assets/a2251270-0c65-4b75-84e7-e450b11bdf99)

And thanks for bringing attention here. I removed the `pages` table as a dependency, so that we don't re-create it. It has to be already available by this time.
Replaced `ctx.ref` with `ctx.resolve` for `FROM all.pages` statement.

Reply:
> Replaced ctx.ref with ctx.resolve for FROM all.pages statement.

Does that mean it would have to return some rows for this to run? Or would this still make it possible to run if `all.pages` exists (which it always will since we never delete it) but doesn't have data for last month?

Reply:
It will read `pages` as required for the SQL for `technologies` table, but don't trigger it as dependency (rebuilding it) in this workflow.

I just don't understand the scenario when we are trying to build cwv tech report table, but all.pages data is not available yet.
Please detail when it happened and how did the timeline look?

Reply:
Covered in https://github.com/HTTPArchive/dataform/pull/3#discussion_r1736956606. Resolving this.